app/src/plugins/remotekeyboard/keyboardlayoutprovider.cpp

```diff
@@ -0,0 +1,259 @@
+                keySequence.append(line);
+            }
+            while (keySequence.contains("\n")) {
+                keySequence.replace("\n", " ");
+            }
+
+            // remove backslash
+            keySequence.replace("\\\"", "\"");
+            keySequence.replace("\\\\", "\\");
+
+            if (keySequence.contains("ShiftKey")) {
+                QJsonObject key;
+                key["caption"] = "shift";
+                row.append(key);
+            } else if (keySequence.contains("BackspaceKey")) {
+                QJsonObject upKey;
+                upKey["caption"] = "";
+                upKey["symView"] = "up";
+                row.append(upKey);
+                QJsonObject key;
+                key["caption"] = "backspace";
+                row.append(key);
+            } else {
+                QRegularExpressionMatch captionMatch =
+                        caption.match(keySequence);
+                QRegularExpressionMatch captionShiftedMatch =
+                        captionShifted.match(keySequence);
+                QRegularExpressionMatch symViewMatch =
+                        symView.match(keySequence);
+                QRegularExpressionMatch symView2Match =
+                        symView2.match(keySequence);
+                QJsonObject key;
+                key["caption"] = captionMatch.captured(1);
+                if (captionShiftedMatch.hasMatch()) {
+                    key["captionShifted"] = captionShiftedMatch.captured(1);
+                }
+                if (symViewMatch.hasMatch()) {
+                    key["symView"] = symViewMatch.captured(1);
+                }
+                if (symView2Match.hasMatch()) {
+                    key["symView2"] = symView2Match.captured(1);
+                }
+                row.append(key);
+            }
+        }
+
+        line = layoutFile.readLine();
+    }
+
+    layoutFile.close();
+
+    m_settings.beginGroup(QStringLiteral("RemoteKeyboard"));
+    m_settings.setValue("layout", layout);
+    m_settings.endGroup();
+
+    m_row2 = keys[0].toArray().toVariantList();
+    m_row3 = keys[1].toArray().toVariantList();
+    m_row4 = keys[2].toArray().toVariantList();
+
+    QJsonArray row5;
+    for (QString key : {"?123", "ctrl", ",", " ", ".", "alt", "enter"}) {
+        QJsonObject keyObject;
+        keyObject["caption"] = key;
+        if (key == ".") {
+            keyObject["symView"] = "left";
+        } else if (key == "alt") {
+            keyObject["symView"] = "down";
+        } else if (key == "enter") {
+            keyObject["symView"] = "right";
+        } else if (key == "?123") {
+            keyObject["symView"] = "ABC";
+        }
+        row5.append(keyObject);
+    }
+
+    m_row5 = row5.toVariantList();
+
+    m_layout = layout;
+
+    emit layoutChanged();
+}
+
+QVariantList KeyboardLayoutProvider::layouts()
+{
+    QDir layoutDir("/usr/share/maliit/plugins/com/jolla/layouts/");
```

Comment:
When parsing "/usr/share/maliit/plugins/com/jolla/layouts*.conf" (QSettings can be used for parsing) you get all layouts with the correct language name.